lib/dns/zone.js

```diff
@@ -0,0 +1,817 @@
+ *     objects to delete from this zone. Be aware that the resource records here
+ *     must match exactly to be deleted.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An API error.
+ * @param {?module:dns/change} callback.change - A {module:dns/change} object.
+ * @param {object} callback.apiResponse - Raw API response.
+ *
+ * @example
+ * var oldARecord = zone.record('a', {
+ *   name: 'example.com.',
+ *   data: '1.2.3.4',
+ *   ttl: 86400
+ * });
+ *
+ * var newARecord = zone.record('a', {
+ *   name: 'example.com.',
+ *   data: '5.6.7.8',
+ *   ttl: 86400
+ * });
+ *
+ * zone.createChange({
+ *   add: newARecord,
+ *   delete: oldARecord
+ * }, function(err, change, apiResponse) {
+ *   if (!err) {
+ *     // The change was created successfully.
+ *   }
+ * });
+ */
+Zone.prototype.createChange = function(options, callback) {
+  var self = this;
+
+  if (!options || !options.add && !options.delete) {
+    throw new Error('Cannot create a change with no additions or deletions.');
+  }
+
+  var body = extend({}, options, {
+    additions: util.arrayize(options.add).map(util.exec('toJSON')),
+    deletions: util.arrayize(options.delete).map(util.exec('toJSON'))
+  });
+
+  delete body.add;
+  delete body.delete;
+
+  var path = '/managedZones/' + this.name + '/changes';
+
+  this.makeReq_('POST', path, null, body, function(err, resp) {
+    if (err) {
+      callback(err, null, resp);
+      return;
+    }
+
+    var change = self.change(resp.id);
+    change.metadata = resp;
+
+    callback(null, change, resp);
+  });
+};
+
+/**
+ * Delete the zone.
+ *
+ * Only empty zones can be deleted. Set options.force to `true` to call
+ * {module:dns/zone#empty} before deleting the zone. Two API calls will then be
+ * made (one to empty, another to delete), which means <strong> this is not an
+ * atomic request</strong>.
+ *
+ * @param {object=} options - Configuration object.
+ * @param {boolean} options.force - Empty the zone before deleting.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An API error.
+ * @param {object} callback.apiResponse - Raw API response.
+ *
+ * @example
+ * zone.delete(function(err, apiResponse) {
+ *   if (!err) {
+ *     // The zone is now deleted.
+ *   }
+ * });
+ *
+ * //-
+ * // Use `force` to first empty the zone before deleting it.
+ * //-
+ * zone.delete({
+ *   force: true
+ * }, function(err, apiResponse) {
+ *   if (!err) {
+ *     // The zone is now deleted.
+ *   }
+ * });
+ */
+Zone.prototype.delete = function(options, callback) {
+  if (util.is(options, 'function')) {
+    callback = options;
+    options = {};
+  }
+
+  if (options.force) {
+    this.empty(this.delete.bind(this, callback));
+    return;
+  }
+
+  var path = '/managedZones/' + this.name;
+  this.makeReq_('DELETE', path, null, null, function(err, resp) {
+    callback(err, resp);
+  });
```

Comment:
Sorry I missed this. The reason is just to limit the arguments that get back to the user's callback. `this.makeReq_` is invoked with `err, resp, parsedResponseBody`, but it's actually not being used anywhere in the library. This question is spawning a new PR (coming soon), so thank you :)